### ISSUE TITLE
Using both non v and v version of git tags

### ISSUE THREAD

**Issue by codvio** [jitpack/jitpack.io#407]
According to semver 1.0.0 git repos must use v prefix for tags.

- http://semver.org/spec/v1.0.0.html#tagging-specification-semvertag

It is removed in version 2 but almost all projects and even semver's itself use v prefix as shown in the URL.

- http://semver.org/spec/v2.0.0.html

Projects use v prefix in git tagging but not in maven repos. For instance this is 1.3.2.RELEASE version of spring-boot project:

- https://github.com/spring-projects/spring-boot/tree/v1.3.2.RELEASE

But you should request non-v version the project:

- http://repo1.maven.org/maven2/org/springframework/boot/spring-boot/1.3.2.RELEASE/

In current jetpack.io version we must use the v version of the project.

Central Repo Version:
```groovy
compile 'org.springframework.boot:spring-boot:1.3.2.RELEASE'
```
JitPack Version:
```groovy
compile 'com.github.spring-projects.spring-boot:spring-boot:v1.3.2.RELEASE'
```

If jitpack can support both v and non-v versions of the projects it can be good.

For instance both lines should be welcome:
```groovy
compile 'com.github.spring-projects.spring-boot:spring-boot:v1.3.2.RELEASE'
compile 'com.github.spring-projects.spring-boot:spring-boot:1.3.2.RELEASE'
```

**Comment by jitpack-io** [comment]
Thank you for the suggestion! In a lot of ways it makes sense but there can be downsides as well:
- First of all it's possible to add two tags - one with v and one without on different commits. 
- If JitPack supports both then its also more confusion for the user - which one should I use? 
- Or when users report bugs you have to try to reproduce it on two versions.

So would recommend using tags without the v because it's non-ambiguous. 

On the other hand, a lot of projects already use the v. Perhaps JitPack should always use a version without the v. When there's only one way to do things then it is simpler and less confusing for everyone.

**Comment by codvio** [comment]
This is the possible pseudo JS code for suggestion, code is the best way to explain something to developers :)

```javascript
function fetchVersion(versionName) {

   if (existInGithub(versionName)) {
      return buildAndDeploy(vesionName, versionName);
   } else if (existInGithub('v' + versionName) {
      return buildAndDeploy('v' + vesionName, versionName);
   }

   return null;
}
```

**Comment by jitpack-io** [comment]
Sure, but the current logic is simpler
```javascript
function fetchVersion(versionName) {
    if (existInGithub(versionName))
        return buildAndDeploy(versionName);
    return null;
}
```
so its easier to explain.

**Comment by raniejade** [comment]
I think you should re-consider this, to be consistent with other libraries deployed in central. Most versions in central don't have a `v` prefix, the only time this is present is on a git tag.

**Comment by jitpack-io** [comment]
@raniejade Agreed, most existing libs don't use the `v` in versions but is that good enough reason to support it considering issues mentioned earlier?
One argument could be that users will intuitively try to specify a version without the `v` and that will fail.

**Comment by raniejade** [comment]
> - First of all it's possible to add two tags - one with v and one without on different commits.
You could enforce it at build time, if a tag looks the same with a previously built one fail the build.

> - If JitPack supports both then its also more confusion for the user - which one should I use?
I don't see any confusion here. People who uses central a lot will intuitively use a version without `v` (a lot of them uses semantic versioning). 

> - Or when users report bugs you have to try to reproduce it on two versions.
How often does that happen? It's the publishers responsibility to ensure that there are no duplicate tags.

**Comment by gdejohn** [comment]
What about using the name of the GitHub release instead of the tag? The API provides both. So you can do v1.0 for the tag to match the git convention, and 1.0 for the GitHub release name and have JitPack use that to match the Maven convention.

**Comment by jitpack-io** [comment]
@gdejohn That sounds very reasonable

**Comment by jitpack-io** [comment]
Ok, versions without 'v' prefix should now be working too

**Comment by gdejohn** [comment]
@jitpack-io, a minor drawback with this approach is that when you go to the JitPack page for a project, the first thing you see is the versions listed under the **Releases** tab, and they correspond to the tags. So for anyone using a dependency from JitPack and omitting the `v` prefix, if the project follows the git tagging convention, it appears to be inconsistent. (I can imagine people wondering, "Wait, that works? Why does that work?")

Have you given any more thought to my proposal?

> What if JitPack used the name of a GitHub release for the version instead of using the tag? The API provides both. So the tag can be `v1.0` to match the git convention, and you can do a GitHub release for that tag and name the release `1.0` to match the Maven convention, and JitPack uses the release name.

Perhaps it should only apply to new releases, and require setting a flag in the `jitpack.yml` configuration (`github_version_release_name: true` or something).

**Comment by jitpack-io** [comment]
Hi @gdejohn ,
Agree that it should only apply to new releases. We still need to figure out how to deal with GitHub API rate limits. We'll quickly exceed if we query releases API or if we check jitpack.yml
